File: src/code-samples/langsmith/smithdb-migration/experiment-runs-query-lookup-experiment-id.kt
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 21
//KOTLIN 2.2.0
//DEPS com.langchain.smith:langsmith-java:0.1.0-beta.13

// :snippet-start: experiment-runs-query-lookup-experiment-id-kt
import com.langchain.smith.client.LangsmithClient
import com.langchain.smith.client.okhttp.LangsmithOkHttpClient
import com.langchain.smith.models.sessions.SessionListParams

// :remove-start:
fun main() {
    if (false) {
// :remove-end:
val client: LangsmithClient = LangsmithOkHttpClient.fromEnv()
val experimentId = client.sessions()
    .list(SessionListParams.builder().name("my-experiment").build())
    .items()
    .first()
    .id()
// :remove-start:
    }
}
// :remove-end:
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/experiment-runs-query-lookup-experiment-id.go
// :snippet-start: experiment-runs-query-lookup-experiment-id-go
package main

import (
	"context"

	"github.com/langchain-ai/langsmith-go"
)

// :remove-start:
func main() {
	if false {
// :remove-end:
ctx := context.Background()
client := langsmith.NewClient()

sessions, err := client.Sessions.List(ctx, langsmith.SessionListParams{
	Name: langsmith.F("my-experiment"),
})
// :remove-start:
if err != nil {
	panic(err.Error())
}
// :remove-end:
experimentID := sessions.Items[0].ID
// :remove-start:
_ = experimentID
	}
}
// :remove-end:
// :snippet-end:
